storage/imx6ull-flash/imx6ull-flash-test.c

```diff
@@ -444,13 +519,166 @@ void test_ecc(void)
 				break;
 			}
 		}
+
+		/* Test assumes no pre-existing bitflips in the page */
+		printf("Checking erased page bitflips handling\n");
+
+		/* Erase the block */
+		err = flashdrv_erase(dma, paddr);
+		if (err < 0) {
+			printf("erase() failed: %d\n", err);
+			break;
+		}
+
+		/* Flip metadata bits (we should correct up to 16 bit flips) */
+		printf("[%4u]: adding bit flips to metadata\n", paddr);
+		err = flip_bits(maxmetaecc, nblock, start, start + rawmetasz, 0);
+		if (err < 0) {
+			break;
+		}
+
+		/* Flip data bits (we should correct up to 14 bit flips) */
+		offs = start + rawmetasz;
+		for (i = 1; i < sizeof(aux->errors); i++) {
+			printf("[%4u]: adding bit flips to data chunk %u\n", paddr, i);
+			err = flip_bits(maxdataecc, nblock, offs, offs + rawdatasz, 0);
+			if (err < 0) {
+				break;
+			}
+			offs += rawdatasz;
+		}
+
+		/* Read the page */
+		err = flashdrv_read(dma, paddr, data, aux);
+		if (err < 0) {
+			printf("read() failed: %d\n", err);
+			break;
+		}
+
+		/* Check and correct bitflips (adapted _flashmtd_checkECC() implementation) */
+		for (i = 0; i < sizeof(aux->errors); i++) {
+			switch (aux->errors[i]) {
+				case flash_no_errors:
+				case flash_erased:
+					break;
+
+				case flash_uncorrectable:
+					/* BCH reports chunk as uncorrectable in case of an erased page with bitflips within that chunk area */
+					/* Check if that's the case by counting the bitflips with an assumption that the page is fully erased */
+
+					/* Read raw page in order to check full chunk - both data and its ECC area */
+					if (raw == NULL) {
+						/* Map raw page buffer */
+						raw = mmap(NULL, pagemapsz, PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_CONTIGUOUS, 0);
+						if (raw == MAP_FAILED) {
+							raw = NULL;
+							err = -ENOMEM;
+							printf("mmap() failed: %d\n", err);
+							break;
+						}
+
+						/* Read raw page */
+						err = flashdrv_readraw(dma, paddr, raw, FLASHDRV_PAGESZ);
+						if (err < 0) {
+							printf("readraw() failed: %d\n", err);
+							break;
+						}
+					}
```

Comment:
As I understand correctly, `raw` should be allocated only once? What do You think about moving it before execution of `for` loop.
Allocating it inside `switch` suggests that it is temporary buffor allocation.